server/functions.lua

```diff
@@ -299,3 +299,17 @@ function QBCore.Functions.IsLicenseInUse(license)
     end
     return false
 end
+
+-- Job Check
+
+function QBCore.Functions.DoesJobExist(job, grade)
+    if job and grade then
+        for k, v in pairs(QBCore.Shared.Jobs) do
```

Comment:
Why are you making this O(n) when it could just be O(1) by using direct lookup?

Reply:
What do you mean?

Reply:
I mean this algorithm you've wrote is O(n) for no reason when you can very easily use an O(1) direct lookup of the table

Reply:
If there is a better way that is better "written" you can do as you wish.

>What is the purpose of this pull request even?
I noticed people asking for it in the Suggestions channel and its something I need too for my staff team.
Its called a "request" for a reason , so... :P

Reply:
This is your pull request so.... it probably won't get merged if you're using algorithms that don't even need to exist since this can be simplified to a simple direct index of the table. There are code guidelines for a reason to ensure that qbcore contains only quality code.